torchrl/modules/models/multiagent.py

```diff
@@ -230,6 +230,221 @@ def forward(self, *inputs: Tuple[torch.Tensor]) -> torch.Tensor:
+        MultiAgentConvNet(
+            (agent_networks): ModuleList(
+                (0-6): 7 x ConvNet(
+                (0): Conv2d(4, 32, kernel_size=(5, 5), stride=(2, 2))
+                (1): ELU(alpha=1.0)
+                (2): Conv2d(32, 32, kernel_size=(5, 5), stride=(2, 2))
+                (3): ELU(alpha=1.0)
+                (4): Conv2d(32, 32, kernel_size=(5, 5), stride=(2, 2))
+                (5): ELU(alpha=1.0)
+                (6): SquashDims()
+                )
+            )
+        )
+        torch.Size([3, 2, 7, 2592])
+        False
+
+        Or where inputs are shared but not parameters.
+        >>> cnn = MultiAgentConvNet(
+        >>>     n_agents,
+        >>>     centralised = True,
+        >>>     share_params = False
+        >>> )
+        >>> print(cnn)
+        >>> print(result.shape)
+        >>> print(all(result[0,0,0] == result[0,0,1]))
+        MultiAgentConvNet(
+            (agent_networks): ModuleList(
+                (0-6): 7 x ConvNet(
+                (0): Conv2d(28, 32, kernel_size=(5, 5), stride=(2, 2))
+                (1): ELU(alpha=1.0)
+                (2): Conv2d(32, 32, kernel_size=(5, 5), stride=(2, 2))
+                (3): ELU(alpha=1.0)
+                (4): Conv2d(32, 32, kernel_size=(5, 5), stride=(2, 2))
+                (5): ELU(alpha=1.0)
+                (6): SquashDims()
+                )
+            )
+        )
+        torch.Size([3, 2, 7, 2592])
+        False
+    """
+
+    def __init__(
+        self,
+        n_agents: int,
+        centralised: bool,
+        share_params: bool,
+        device: Optional[DEVICE_TYPING] = None,
+        num_cells: Optional[Sequence[int]] = None,
+        kernel_sizes: Union[Sequence[Union[int, Sequence[int]]], int] = 5,
+        strides: Union[Sequence, int] = 2,
+        paddings: Union[Sequence, int] = 0,
+        activation_class: Type[nn.Module] = nn.ELU,
+        **kwargs,
+    ):
+        super().__init__()
+
+        self.n_agents = n_agents
+        self.centralised = centralised
+        self.share_params = share_params
+
+        self.agent_networks = nn.ModuleList(
+            [
+                ConvNet(
+                    num_cells=num_cells,
+                    kernel_sizes=kernel_sizes,
+                    strides=strides,
+                    paddings=paddings,
+                    activation_class=activation_class,
+                    device=device,
+                    **kwargs,
+                )
+                for _ in range(self.n_agents if not self.share_params else 1)
+            ]
+        )
+
+    def forward(self, inputs: torch.Tensor):
+        assert (
+            len(inputs.shape) >= 4
+        ), """Multi-agent network expects" (*batch_size, agent_index, x, y, channels)"""
+        assert (
+            inputs.shape[-4] == self.n_agents
+        ), f"""Multi-agent network expects {self.n_agents} but got {inputs.shape[-4]}"""
+        # If the model is centralized, agents have full observability
```

Comment:
We don't use assert in the main code base.
This should be a ValueError I guess?